pkg/hypervctl/disk_resize.go

```diff
@@ -0,0 +1,22 @@
+package hypervctl
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/containers/podman/v4/pkg/strongunits"
+)
+
+// ResizeDisk takes a diskPath and strongly typed new size and uses powershell
+// to change its size.  There is no error protection for trying to size a disk
+// smaller than the current size.
+func ResizeDisk(diskPath string, newSize strongunits.GiB) error {
+	resize := exec.Command("powershell", []string{"-command", fmt.Sprintf("Resize-VHD %s %d", diskPath, newSize.ToBytes())}...)
```

Comment:
Note there is also an API call you can make on the ImageManagementService, similar to how the CreateVhdxFile function does, but instead against the resize method:

https://learn.microsoft.com/en-us/windows/win32/hyperv_v2/resizevirtualharddisk-msvm-imagemanagementservice

Reply:
Code would look something like this:
```go
        err = imms.BeginInvoke("ResizeVirtualHardDisk").
		In("Path", diskPath).
                In("MaxInternalSize", newSize.ToBytes()).
		Execute().
		Out("Job", &job).
		Out("ReturnValue", &ret).
		End()

	if err != nil {
		return fmt.Errorf("failed to resize disk: %w", err)
	}

	return waitVMResult(ret, service, job, "failed to resize disk", nil)
```